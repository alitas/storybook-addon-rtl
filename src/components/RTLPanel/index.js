import React, { Component } from 'react'
import PropTypes from 'prop-types'
import RTLToggle from '../RTLToggle'
import { Panel } from './styles'

export default class RTLPanel extends Component {
  render () {
    const { checked, onChange } = this.props

    return (
      <Panel>
        <RTLToggle
          checked={checked}
          onChange={onChange}
        />
      </Panel>
    )
  }
}

RTLPanel.propTypes = {
  checked: PropTypes.bool,
  onChange: PropTypes.func.isRequired
}

RTLPanel.defaultProps = {
  checked: false
}
