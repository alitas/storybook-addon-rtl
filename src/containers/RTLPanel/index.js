import React, { Component } from 'react'
import PropTypes from 'prop-types'
import RTLPanelComponent from '../../components/RTLPanel'
import { getDefaultTextDirection } from '../../utils'
import { INITIALIZE_EVENT_ID, UPDATE_EVENT_ID } from '../../constants'

export default class RTLPanel extends Component {
  constructor (props) {
    super(props)

    this.state = {
      direction: getDefaultTextDirection(props.api)
    }

    this.handleInitializeEvent = this.emitUpdateEvent.bind(this)
    this.handlePanelChange = this.handlePanelChange.bind(this)
  }

  componentDidMount () {
    this.props.channel.on(INITIALIZE_EVENT_ID, this.emitUpdateEvent)
  }

  componentWillUnmount () {
    this.props.channel.removeListener(INITIALIZE_EVENT_ID, this.emitUpdateEvent)
  }

  componentDidUpdate () {
    this.emitUpdateEvent()
  }

  emitUpdateEvent () {
    this.props.channel.emit(UPDATE_EVENT_ID, this.state)
  }

  handlePanelChange (checked) {
    const state = { direction: checked ? 'rtl' : 'ltr' }
    this.setState(state)
  }

  render () {
    const { direction } = this.state

    return (
      <RTLPanelComponent
        checked={direction === 'rtl'}
        onChange={this.handlePanelChange}
      />
    )
  }
}

RTLPanel.propTypes = {
  api: PropTypes.object.isRequired,
  channel: PropTypes.object.isRequired
}
