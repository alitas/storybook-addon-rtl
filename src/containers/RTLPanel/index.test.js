import React from 'react'
import { shallow, configure } from 'enzyme'
import RTLPanel from '.'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('RTLPanel', () => {
  let api
  let channel
  let wrapper

  beforeEach(() => {
    api = {
      getQueryParam: jest.fn()
    }
    channel = {
      on: jest.fn(),
      removeListener: jest.fn(),
      emit: jest.fn()
    }
  })

  describe('without query parameter', () => {
    beforeEach(() => {
      api.getQueryParam.mockReturnValue(undefined)
      wrapper = shallow(<RTLPanel api={api} channel={channel} />)
    })

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with query parameter', () => {
    beforeEach(() => {
      api.getQueryParam.mockReturnValue('rtl')
      wrapper = shallow(<RTLPanel api={api} channel={channel} />)
    })

    it('renders', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
