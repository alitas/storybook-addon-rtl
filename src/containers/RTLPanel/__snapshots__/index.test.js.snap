// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RTLPanel with direction set while panel closed renders 1`] = `
<div
  class="sc-jSMfEi hfxUgk"
>
  <span
    class="sc-bczRLJ kxUEBA"
  >
    <label
      class="sc-gsnTZi ecyAuL"
    >
      <input
        aria-controls="rtl-status"
        checked=""
        class="sc-dkzDqf khKIDk"
        data-testid="rtl-toggle-input"
        type="checkbox"
      />
      <a
        class="sc-hKMtZM kLroqp"
      >
        Enable right-to-left
      </a>
    </label>
    <span
      aria-live="polite"
      class="sc-eCYdqJ gVppkG"
      id="rtl-status"
      role="region"
    >
      Right-to-left
    </span>
  </span>
</div>
`;

exports[`RTLPanel with query parameter renders 1`] = `
<div
  class="sc-jSMfEi hfxUgk"
>
  <span
    class="sc-bczRLJ kxUEBA"
  >
    <label
      class="sc-gsnTZi ecyAuL"
    >
      <input
        aria-controls="rtl-status"
        checked=""
        class="sc-dkzDqf khKIDk"
        data-testid="rtl-toggle-input"
        type="checkbox"
      />
      <a
        class="sc-hKMtZM kLroqp"
      >
        Enable right-to-left
      </a>
    </label>
    <span
      aria-live="polite"
      class="sc-eCYdqJ gVppkG"
      id="rtl-status"
      role="region"
    >
      Right-to-left
    </span>
  </span>
</div>
`;

exports[`RTLPanel with story param renders 1`] = `
<div
  class="sc-jSMfEi hfxUgk"
>
  <span
    class="sc-bczRLJ kxUEBA"
  >
    <label
      class="sc-gsnTZi ecyAuL"
    >
      <input
        aria-controls="rtl-status"
        checked=""
        class="sc-dkzDqf khKIDk"
        data-testid="rtl-toggle-input"
        type="checkbox"
      />
      <a
        class="sc-hKMtZM kLroqp"
      >
        Enable right-to-left
      </a>
    </label>
    <span
      aria-live="polite"
      class="sc-eCYdqJ gVppkG"
      id="rtl-status"
      role="region"
    >
      Right-to-left
    </span>
  </span>
</div>
`;

exports[`RTLPanel with story param responds to changes 1`] = `
<div
  class="sc-jSMfEi hfxUgk"
>
  <span
    class="sc-bczRLJ kxUEBA"
  >
    <label
      class="sc-gsnTZi ecyAuL"
    >
      <input
        aria-controls="rtl-status"
        checked=""
        class="sc-dkzDqf khKIDk"
        data-testid="rtl-toggle-input"
        type="checkbox"
      />
      <a
        class="sc-hKMtZM kLroqp"
      >
        Enable right-to-left
      </a>
    </label>
    <span
      aria-live="polite"
      class="sc-eCYdqJ gVppkG"
      id="rtl-status"
      role="region"
    >
      Right-to-left
    </span>
  </span>
</div>
`;

exports[`RTLPanel with story param responds to changes 2`] = `
<div
  class="sc-jSMfEi hfxUgk"
>
  <span
    class="sc-bczRLJ kxUEBA"
  >
    <label
      class="sc-gsnTZi isuMOS"
    >
      <input
        aria-controls="rtl-status"
        checked=""
        class="sc-dkzDqf khKIDk"
        data-testid="rtl-toggle-input"
        type="checkbox"
      />
      <a
        class="sc-hKMtZM jsWbyR"
      >
        Enable right-to-left
      </a>
    </label>
    <span
      aria-live="polite"
      class="sc-eCYdqJ gVppkG"
      id="rtl-status"
      role="region"
    >
      Left-to-right
    </span>
  </span>
</div>
`;

exports[`RTLPanel without query parameter renders 1`] = `
<div
  class="sc-jSMfEi hfxUgk"
>
  <span
    class="sc-bczRLJ kxUEBA"
  >
    <label
      class="sc-gsnTZi isuMOS"
    >
      <input
        aria-controls="rtl-status"
        class="sc-dkzDqf khKIDk"
        data-testid="rtl-toggle-input"
        type="checkbox"
      />
      <a
        class="sc-hKMtZM jsWbyR"
      >
        Enable right-to-left
      </a>
    </label>
    <span
      aria-live="polite"
      class="sc-eCYdqJ gVppkG"
      id="rtl-status"
      role="region"
    >
      Left-to-right
    </span>
  </span>
</div>
`;
